### PR TITLE
 deps(jasmine): upgrade jasmine 3.3

### PR DESCRIPTION
**Pull request by cnishina**

### CODE REVIEW

**Comment by cnishina** [comment]
This PR is chained to https://github.com/angular/protractor/pull/5101. So the commit that should be reviewed is this one: https://github.com/angular/protractor/pull/5102/commits/e6d3939a076ebdf13a1e8f6b213d56e9565387a3

File: spec/basic/expected_conditions_spec.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
-const EC = protractor.ExpectedConditions;
-
 describe('expected conditions', () => {
+  let EC = null;
```
Comment (heathkit) [inline spec/basic/expected_conditions_spec.js]: Why do we need to move this into a beforeEach?

File: lib/expectedConditions.ts [hunk 2]
```diff
@@ -278,12 +280,15 @@ export class ProtractorExpectedConditions {
...
-  titleIs(title: string): Function {
-    return () => {
-      return this.browser.driver.getTitle().then((actualTitle: string): boolean => {
-        return actualTitle === title;
-      });
+  titleIs(title: string): (() => Promise<boolean>) {
+    return async(): Promise<boolean> => {
+      const actualTitle = await this.browser.driver.getTitle();
+      return actualTitle === title;
+<<<<<<< HEAD
```
Comment (heathkit) [inline lib/expectedConditions.ts]: Conflict markers

File: lib/expectedConditions.ts [hunk 3]
```diff
@@ -205,13 +201,16 @@ export class ProtractorExpectedConditions {
...
+  textToBePresentInElement(elementFinder: ElementFinder, text: string): (() => Promise<boolean>) {
+    let hasText = async () => {
+      try {
+        const actualText = await elementFinder.getText();
         // MSEdge does not properly remove newlines, which causes false
         // negatives
         return actualText.replace(/\r?\n|\r/g, '').indexOf(text) > -1;
-      }, falseIfMissing);
+      } catch (e) {
+        falseIfMissing(e);
```
Comment (heathkit) [inline lib/expectedConditions.ts]: Should return

File: lib/browser.ts [hunk 4]
```diff
@@ -838,7 +838,7 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
 
     // Reset bpClient sync
     if (this.bpClient) {
-      await this.bpClient.setWaitEnabled(!this.internalIgnoreSynchronization);
+      await this.bpClient.setWaitEnabled(await this.waitForAngularEnabled());
```
Comment (heathkit) [inline lib/browser.ts]: I think this change is also in the other PR - it's ok to use internalIgnoreSynchronization here

File: lib/expectedConditions.ts (hunk 3 above)
Comment (cnishina) [reply lib/expectedConditions.ts]: Fixed in a different PR.

**Comment by cnishina** [comment]
Not working because it still relies on the previous PR to be approved. I will rebase it later.

**Comment by cnishina** [comment]
Rebasing...

**Comment by praneethmendu** [comment]
Can someone please add some reference to why it was rolled back to 2.8 and how, if it is even possible, to work with 3.3 now ?

**Pull request merged**